source/common/router/config_impl.cc

```diff
@@ -511,21 +520,39 @@ RouteEntryImplBase::loadRuntimeData(const envoy::config::route::v3::RouteMatch&
   return runtime;
 }
 
+// finalizePathHeaders does the "standard" path rewriting, meaning that it
+// handles the "prefix_rewrite" and "regex_rewrite" route actions, only one of
+// which can be specified. The "matched_path" argument applies only to the
+// prefix rewriting, and describes the portion of the path (excluding query
+// parameters) that should be replaced by the rewrite. A "regex_rewrite"
+// applies to the entire path (excluding query parameters), regardless of what
+// portion was matched.
 void RouteEntryImplBase::finalizePathHeader(Http::HeaderMap& headers,
                                             absl::string_view matched_path,
                                             bool insert_envoy_original_path) const {
   const auto& rewrite = getPathRewrite();
-  if (rewrite.empty()) {
+  if (!rewrite.empty()) {
+    std::string path(headers.Path()->value().getStringView());
+    if (insert_envoy_original_path) {
+      headers.setEnvoyOriginalPath(path);
+    }
```

Comment:
I think refactoring this to avoid repeating the initialization of `path` and handling of `insert_envoy_original_path` would be better. It seems inevitable that there will be another path manipulation method in the future.

I'm thinking short-circuit on empty `rewrite` and null `regex_rewrite_` at the top, init `path` and handle `insert_envoy_orginal_path`, and then handle rewrite else handle rewrite regex.

Reply:
Changed as suggested.  I considered doing it that way originally, but didn't since it was a slightly bigger change to the `prefix_write` path.  But that was a silly reason, I think.